src/charm.py

```diff
@@ -109,11 +150,34 @@ def _install_cli_resources(self):
             log.exception("CLI binaries could not be installed.")
             return
 
+    def _install_service(self, service_file_path):
+        try:
+            service_path = Path("/etc/systemd/system")
+            shutil.copy(service_file_path, service_path)
+            subprocess.run(["systemctl", "daemon-reload"])
```

Comment:
`run(...)` doesn't raise a `CalledProcessError` without `check=True`

```suggestion
            subprocess.check_call(["systemctl", "daemon-reload"])
```

here's an example
```python
>>> subprocess.run(["systemctl", "blah"]).returncode
Unknown command verb blah.
1
>>> subprocess.check_call(["systemctl", "blah"])
Unknown command verb blah.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['systemctl', 'blah']' returned non-zero exit status 1.
```